sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statsEstimation/EstimationUtils.scala

```diff
@@ -80,6 +80,54 @@ object EstimationUtils {
     expressions.collect {
       case alias @ Alias(attr: Attribute, _) if attributeStats.contains(attr) =>
         alias.toAttribute -> attributeStats(attr)
+      case alias @ Alias(expn: Expression, _) if isExpressionStatsExist(expn, attributeStats) =>
```

Comment:
Why did you update this method instead of `AggregateEstimation`? `Project` uses this method though. Is this related to projections?

Reply:
Yes, We need to update for project node as well. If the grouping key is aliased, or if the grouping key is complex expression, the expressions are pushed to project node.
Even if the aggregate node is not there, still we need to update the stats for expressions in the project node.

Reply:
Ah I see.